etc/docs/generate-docs.ts

```diff
@@ -0,0 +1,150 @@
+#! /usr/bin/env ts-node
+
+import { parse, stringify } from '@iarna/toml';
+import { exec as execCb } from 'child_process';
+import { readFileSync, writeFile as writeFileCb } from 'fs';
+import { promisify } from 'util';
+import { createInterface } from 'readline';
+import { chdir } from 'process';
+
+const exec = promisify(execCb);
+const writeFile = promisify(writeFileCb);
+
+const RELEASES_TOML_FILE = './template/data/releases.toml';
+const RELEASES_JSON_FILE = './template/static/versions.json';
+
+interface JsonVersionSchema {
+  version: string;
+}
+
+interface VersionSchema {
+  version: string;
+  status: string;
+  api: string;
+  usesMongoDBManual?: boolean;
+  docs?: string;
+  semverVersion: string;
+}
+
+interface TomlVersionSchema {
+  current: string;
+  mongodDBManual: string;
+  versions: VersionSchema[]
+}
+
+function prompt(prompt: string) : Promise<string> {
+  const rl = createInterface({
+    input: process.stdin,
+    output: process.stderr
+  })
+
+  return new Promise((resolve, _) => {
+    rl.question(prompt, (answer) => {
+      rl.close();
+      resolve(answer);
+    })
+  })
+}
+
+function getCommandLineArguments() : { semverVersion: string, status: string } {
+  const args = process.argv.slice(2);
+
+  if (args.length === 0) {
+    console.error('usage: generate-docs.ts <semver version> <status (optional)>')
+    process.exit(1);
+  }
+
+  const semverVersion = args.shift();
+  const status = args.shift() ?? 'current';
+  return {
+    semverVersion,
+    status
+  }
+}
+
+async function copyNewDocsToGeneratedSite({ semverVersion }: VersionSchema) {
+  const outputDirectory = `./temp/${semverVersion}`;
+  const pathToBuiltDocs = './build';
+  const command = `cp -R ${pathToBuiltDocs} ${outputDirectory}`;
+  return await exec(command);
+}
+
+async function updateSiteTemplateForNewVersion(newVersion: VersionSchema, tomlData: TomlVersionSchema, jsonVersions: JsonVersionSchema[]) {
+  const versionExists = jsonVersions.some(({ version }) => version === newVersion.semverVersion);
+  if (versionExists) {
+    const existingVersionIndex = tomlData.versions.findIndex(({ semverVersion }) => semverVersion === newVersion.semverVersion);
+    tomlData.versions[existingVersionIndex] = newVersion;
+  } else {
+    tomlData.versions.unshift(newVersion);
+    tomlData.current = newVersion.version;
+
+    jsonVersions.unshift({ version: newVersion.semverVersion })
```

Comment:
I think we could make this process a bit more resilient to generating versions out of order by just writing a custom semver sort function and passing it to both tomlData and and jsonVersions (should be straightforward since it's internal and also we only do major.minor, no patch or pre-release flags); we could also have it check for our special keyword `main` or `current`, something like:
```js
const versionSort = (a, b) => {
  const vA = a.version;
  const vB = b.version;
  if (vA === 'current') { return -1; }
  if (vB === 'current') { return 1; }
//  < logic to return -1 if vA > vB and 1 if vA < bB >
// since we want the most recent version first
}

<unshift new version>
tomlData.versions.sort(versionSort);
tomlData.current = tomlData.versions[0].versions;

<unshift new version>
jsonVersions.sort(versionSort);

```